### ISSUE TITLE
Issue with Platform Average

### ISSUE THREAD

**Issue by simonjowett** [DavidFW1960/bom-weather-card#33]
Hi David - on updating my configurations I'm coming up against an issue with this section of code:

```  - platform: average
    name: today_temp_bom
    entities:
      - sensor.gosford_temperature
    start: '{{ now().replace(hour=0).replace(minute=0).replace(second=0) }}'
    end: '{{ now() }}'


      bom_today_max:
        value_template: >
          {%- if states('sensor.kariong_max_0') == 'n/a' -%} 
            {{ state_attr('sensor.today_temp_bom', 'max_value') }}
          {% else %}
            {{ states('sensor.kariong_max_0') }}
          {% endif %}
      bom_today_min:
        value_template: >
          {%- if states('sensor.sensor.kariong_min_0') == 'n/a' -%} 
            {{ state_attr('sensor.today_temp_bom', 'min_value') }}
          {% else %}
            {{ states('sensor.kariong_min_0') }}
          {% endif %}
```

this throws up a indenting error
![image](https://user-images.githubusercontent.com/13506791/97643933-0b7f1e80-1a9d-11eb-90c6-6894dd331021.png)

If I move the bom_today_max back 1 tab the error goes away in the file editor but trying "check configuration" fails

![image](https://user-images.githubusercontent.com/13506791/97644073-6b75c500-1a9d-11eb-970b-30f51b899708.png)

I think I've installed the "average" integration properly  - any ideas?

**Comment by DavidFW1960** [comment]
did you change the gosford to your sensor for sensor.gosford_temperature ?
does sensor.today_temp_bom exist?

Also did you restart after you added the custom average component? You will have to disable those lines in the package before you restart too because it will fail the config check until the component is loaded....

**Comment by simonjowett** [comment]
I've also tried with my local suburb name 'tapitallee' it still gives the same error.

sensor.today_temp_bom does exist but it is unavailable - perhaps this is the issue?

![image](https://user-images.githubusercontent.com/13506791/97646710-72540600-1aa4-11eb-8dd4-bbfde59f98a6.png)

As to your last point I only added the average component AFTER.  However, I've now restarted successfully but the problem remains.  I can get rid of the problem by just removing this "-platform average" section.  The lovelace card still seems to work ok?!

**Comment by simonjowett** [comment]
Is this line (190) correct?   'sensor.sensor.kariong_min_0'

```          {%- if states('sensor.sensor.kariong_min_0') == 'n/a' -%} 
```

**Comment by simonjowett** [comment]
FYI - this is my location specific config

```  - platform: average
    name: today_temp_bom
    entities:
      - sensor.nowra_temperature
    start: '{{ now().replace(hour=0).replace(minute=0).replace(second=0) }}'
    end: '{{ now() }}'


      bom_today_max:
        value_template: >
          {%- if states('sensor.tapitallee_max_0') == 'n/a' -%} 
            {{ state_attr('sensor.today_temp_bom', 'max_value') }}
          {% else %}
            {{ states('sensor.tapitallee_max_0') }}
          {% endif %}
      bom_today_min:
        value_template: >
          {%- if states('sensor.tapitallee_min_0') == 'n/a' -%} 
            {{ state_attr('sensor.today_temp_bom', 'min_value') }}
          {% else %}
            {{ states('sensor.tapitallee_min_0') }}
          {% endif %}
```

**Comment by DavidFW1960** [comment]
> Is this line (190) correct? 'sensor.sensor.kariong_min_0'
> 
> ` {%- if states('sensor.sensor.kariong_min_0') == 'n/a' -%}`

dammit no it's not....
> ` {%- if states('sensor.kariong_min_0') == 'n/a' -%}`

so that code is under:
sensor:
  - platform: average

and the bom_today is a platform: template

```
sensor:
  - platform: template
    sensors:
      bom_today_max:
        value_template: >
          {{ state_attr('sensor.today_temp_bom', 'max_value') }}

      bom_today_min:
        value_template: >
          {{ state_attr('sensor.today_temp_bom', 'min_value') }}

  - platform: average
    name: today_temp_bom
    entities:
      - sensor.nowra_temperature
    start: '{{ now().replace(hour=0).replace(minute=0).replace(second=0) }}'
    end: '{{ now() }}'
```
NOTE there are 2 kinds of sensors average and template and if you have shown your correct config what you have shown there for sure will not work....

check sensor.nowra_temperature exists
then check sensor.today_temp_bom exists and it has min_value and max_value attributes.
Then as long as the bom_today_max or min are in the template section as per the above they should exist.

**Comment by DavidFW1960** [comment]
I don't even know where you found this... it's not in the docs I can see?
```{%- if states('sensor.sensor.kariong_min_0') == 'n/a' -%}```

**Comment by DavidFW1960** [comment]
oh never mind it's in the package. DOH!!

**Comment by DavidFW1960** [comment]
and I see I have both average and statistics in the package too. Only 1 can be used. Sorry about that......

**Comment by DavidFW1960** [comment]
I have updated the package now

**Comment by simonjowett** [comment]
ok by adding 2 lines
 - platform template
     sensors:

it now works but your .yaml file seems to have these 2 lines missing

![image](https://user-images.githubusercontent.com/13506791/97649743-3886fd80-1aac-11eb-8638-cb801f4d7d3c.png)

**Comment by DavidFW1960** [comment]
Really sorry. I just updated the package again.
There's a few ways to do the today_temp_bom - either the average or statistics. (Average is best)
With the new BOM integration, the min and max don't disappear like the old ones did (well the min seems to be deleted in the afternoon) So I changed the way it does the min and max and the old version (which you are using) is in the card right now.
A consequence of the change is in my card on my HA, the min and max I pass through to the card are always the ACTUAL not forecast for today using that average sensor.

Anyway, thanks for beta testing my new docs lol!
